assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.js

```diff
@@ -306,14 +226,50 @@ const SearchFunnelWidget = ( { Widget, WidgetReportError } ) => {
 			);
 		}
 	);
-	const analyticsVisitorsOverviewAndStatsError = useSelect( ( select ) => {
-		if ( ! isAnalyticsConnected || showRecoverableAnalytics ) {
+
+	const analyticsLoading = useSelect( ( select ) => {
+		if (
+			! isAnalyticsConnected ||
+			! canViewSharedAnalytics ||
+			showRecoverableAnalytics
+		) {
 			return false;
 		}
 
-		return select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [
-			analyticsVisitorsOverviewAndStatsArgs,
-		] );
+		return (
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsOverviewArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsStatsArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [
+				analyticsVisitorsOverviewAndStatsArgs,
+			] ) ||
+			! select( MODULES_ANALYTICS ).hasFinishedResolution(
+				'getGoals',
+				[]
+			)
+		);
```

Comment:
This works, but we can make it a bit more readable:

```suggestion
		const { hasFinishedResolution } = select( MODULES_ANALYTICS );

		return ! (
			hasFinishedResolution( 'getReport', [ analyticsOverviewArgs ] ) &&
			hasFinishedResolution( 'getReport', [ analyticsStatsArgs ] ) &&
			hasFinishedResolution( 'getReport', [
				analyticsVisitorsOverviewAndStatsArgs,
			] ) &&
			hasFinishedResolution( 'getGoals', [] )
		);
```